test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +93,11 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        } else {
+            assertEquals(new AddressBook(), saveFile.load());
```

Comment:
:+1: for covering both cases.
However, you can also check if the in memory data is the same as expected and not modified.